scripts/templates/ldrddi.cpp.mako

```diff
@@ -154,13 +164,27 @@ namespace loader
             // check if the arg value is a translated handle
             ze_image_object_t **imageHandle = static_cast<ze_image_object_t **>(internalArgValue);
             ze_sampler_object_t **samplerHandle = static_cast<ze_sampler_object_t **>(internalArgValue);
-            if( context->image_handle_map.find(*imageHandle) != context->image_handle_map.end() ) {
-                internalArgValue = &context->image_handle_map[*imageHandle];
-            } else if( context->sampler_handle_map.find(*samplerHandle) != context->sampler_handle_map.end() ) {
-                internalArgValue = &context->sampler_handle_map[*samplerHandle];
+            {
+                std::lock_guard<std::mutex> image_lock(context->image_handle_map_lock);
+                std::lock_guard<std::mutex> sampler_lock(context->sampler_handle_map_lock);
+                if( context->image_handle_map.find(*imageHandle) != context->image_handle_map.end() ) {
+                    internalArgValue = &context->image_handle_map[*imageHandle];
+                } else if( context->sampler_handle_map.find(*samplerHandle) != context->sampler_handle_map.end() ) {
+                    internalArgValue = &context->sampler_handle_map[*samplerHandle];
+                }
             }
         }
         %endif
+        %if re.match(r"\w+CommandListAppendMetricQueryEnd$", th.make_func_name(n, tags, obj)):
```

Comment:
Can you add a comment here explaining that this is a temporary workaround?

Reply:
added a comment